CONTRIBUTING.md

```diff
@@ -50,9 +50,101 @@ gpuCI.
 
 There is one current exception `/legacy` which hosts legacy images.
 
-## Extending Containers
+## Extending Images
 
-More information to come...
+### How can I create images to build my project on gpuCI?
+
+gpuCI can use any docker images to build your project. However, the RAPIDS Ops team has some prebuilt parent images which make things easier especially if you depend on CUDA and Conda.
+
+See [Public Images](README.md#public-images) for list of options for `FROM`-ing
+
+Here is an example on extending the image:
+
+```Dockerfile
+ARG FROM_IMAGE=gpuci/miniconda-cuda
+ARG CUDA_VERSION=9.2
+ARG CUDA_VER=${CUDA_VERSION}
+ARG CUDA_TYPE=devel
+ARG LINUX_VERSION=ubuntu16.04
+FROM ${FROM_IMAGE}:${CUDA_VERSION}-${CUDA_TYPE}-${LINUX_VERSION}
+
+# Define arguments
+ARG CUDA_VER
+ARG PYTHON_VERSION=3.6
+ARG LIB_NG_VERSION=7.3.0
+
+# Set environment
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/lib
+ENV PATH=${PATH}:/conda/bin
+
+# Create a dev conda env
+RUN source activate base \
+    && conda create --no-default-packages --override-channels -n dev \
+      -c nvidia \
+      -c conda-forge \
+      -c defaults \
+      nomkl \
+      cudatoolkit=${CUDA_VER} \
+      conda-forge::blas=1.1=openblas \
+      libgcc-ng=${LIB_NG_VERSION} \
+      libstdcxx-ng=${LIB_NG_VERSION} \
+      python=${PYTHON_VERSION} \
+      #TODO: Add additional conda packages here
+    && conda clean -afy \
+    && sed -i 's/conda activate base/conda activate dev/g' ~/.bashrc \
+    && chmod -R ugo+w /opt/conda
+
+## Enables "source activate conda"
+SHELL ["/bin/bash", "-c"]
+
+ENTRYPOINT [ "/usr/bin/tini", "--" ]
+CMD [ "/bin/bash" ]
+```
+
+If you can include all of the conda dependencies in the image, building the actual project will be much faster.
+
+
+### What name and tag do I use?
+
+`gpuci/<project>:${PROJECT_VERSION}-cuda${CUDA_VERSION}-devel-${LINUX_VERSION}-py${PYTHON_VERSION}`
+
+Example: `gpuci/rapidsai:0.15-cuda10.1-devel-ubuntu18.04-py3.6`
+
+
+### How can I use gcc7 in centos7 images?
+
+Simply include this snippet in a Dockerfile.centos7 file. You can use the same template above, replacing the 'Set environment section'
+
+```
+# Install gcc7 from prebuilt image
```

Comment:
This should be updated to use the tarball method used in https://github.com/raydouglass/gpuci-build-environment/blob/extending-docs/rapidsai/devel-centos7.Dockerfile#L83 the env vars are still needed and are set further up in the same file